solidity/test/isms/CCIPIsm.t.sol

```diff
@@ -0,0 +1,480 @@
+    uint64 internal constant OPTIMISM_CHAIN_SELECTOR = 3734403246176062136;
+
+    event MessageSent(
+      bytes32 indexed messageId, // The unique ID of the CCIP message.
+      uint64 indexed destinationChainSelector, // The chain selector of the destination chain.
+      address receiver, // The address of the receiver on the destination chain.
+      bytes callData, // The payload being sent
+      address feeToken, // the token address used to pay CCIP fees.
+      uint256 fees // The fees paid for sending the CCIP message.
+    );
+
+    // Event emitted when a message is received from another chain.
+    event MessageReceived(
+      bytes32 indexed messageId, // The unique ID of the CCIP message.
+      uint64 indexed sourceChainSelector, // The chain selector of the source chain.
+      address sender, // The address of the sender from the source chain.
+      bytes32 payload // The payload that was received.
+    );
+
+    event ReceivedMessage(bytes32 indexed messageId);
+
+    function setUp() public {
+        // block numbers to fork from, chain data is cached to ../../forge-cache/
+        mainnetFork = vm.createFork(vm.rpcUrl("mainnet"), 18_787_859);
+        optimismFork = vm.createFork(vm.rpcUrl("optimism"), 106_233_774);
+
+        testRecipient = new TestRecipient();
+
+        encodedMessage = _encodeTestMessage();
+        messageId = Message.id(encodedMessage);
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    ///                            SETUP                            ///
+    ///////////////////////////////////////////////////////////////////
+
+    function deployCCIPHookAndISMMainnet() public {
+        vm.selectFork(mainnetFork);
+
+        l1Mailbox = new TestMailbox(MAINNET_DOMAIN);
+
+        ccipHookMainnet = new CCIPHook(
+            MAINNET_ROUTER_ADDRESS,
+            address(l1Mailbox)
+        );
+
+        ccipHookMainnet.addDestinationChainToAllowlist(OPTIMISM_CHAIN_SELECTOR, true);
+
+        vm.makePersistent(address(ccipHookMainnet), MAINNET_ROUTER_ADDRESS);
+    }
+
+    function deployCCIPHookAndISMOptimism() public {
+        vm.selectFork(optimismFork);
+
+        l2Mailbox = new TestMailbox(MAINNET_DOMAIN);
+
+        ccipISMOptimism = new CCIPIsm(OP_ROUTER_ADDRESS);
+
+        ccipISMOptimism.addSourceChainToAllowlist(MAINNET_CHAIN_SELECTOR, true);
+        ccipISMOptimism.setAuthorizedHook(TypeCasts.addressToBytes32(address(OP_ROUTER_ADDRESS)));
+
+        vm.makePersistent(address(ccipISMOptimism), OP_ROUTER_ADDRESS);
+    }
+
+    function deployAll() public {
+        deployCCIPHookAndISMMainnet();
+        deployCCIPHookAndISMOptimism();
+
+        vm.selectFork(optimismFork);
+        ccipISMOptimism.addSenderToAllowlist(address(ccipHookMainnet), true);
+
+        // for sending value
+        vm.deal(
+            OP_ROUTER_ADDRESS,
+            2 ** 255
+        );
+    }
+
+    ///////////////////////////////////////////////////////////////////
```

Comment:
useful to add test for updating allowlist and checking when allowlisted != [authorizedHook]

Reply:
Based on my answer above, I'm not sure what has to be done, but maybe I'm missing something

The check is done [here](https://github.com/Raid5594/hyperlane-monorepo/blob/cae554ffaf4f02e0a21870766a08ae8b2a3638d6/solidity/test/isms/CCIPIsm.t.sol#L297)